akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpHeaderParser.scala

```diff
@@ -544,8 +553,21 @@ private[http] object HttpHeaderParser {
                 case -1 ⇒ if (sb != null) sb.append(c).append(byteChar(input, ix + 1)).append(byteChar(input, ix + 2)).append(byteChar(input, ix + 3)) else null
                 case cc ⇒ appended2(cc)
               }
-            } else fail(s"Illegal character '${escape(c)}' in header value")
-          scanHeaderValue(hhp, input, start, limit)(nsb, nix)
+            } else {
+              mode match {
+                case ParserSettings.IllegalResponseHeaderValueProcessingMode.Error ⇒
+                  fail(s"Illegal character '${escape(c)}' in header value")
+                case ParserSettings.IllegalResponseHeaderValueProcessingMode.Warn ⇒
+                  // ignore the illegal character and log a warning message
+                  log.warning(s"Illegal character '${escape(c)}' in header value")
+                  sb
+                case ParserSettings.IllegalResponseHeaderValueProcessingMode.Ignore ⇒
+                  // just ignore the illegal character
+                  sb
+              }
```

Comment:
I see what you suggest here but not sure it's worth the effort. I don't think we want to allow an open-world for other handlers here, just the defined within Akka ones are good enough, so OK to keep here.